frontend/src/metabase-types/api/settings.ts

```diff
@@ -191,54 +191,62 @@ export interface OpenAiModel {
 export type HelpLinkSetting = "metabase" | "hidden" | "custom";
 
 interface InstanceSettings {
-  "active-users-count"?: number;
   "admin-email": string;
-  "bcc-enabled?": boolean;
-  "deprecation-notice-version"?: string;
-  "ee-openai-api-key"?: string;
-  "embedding-secret-key"?: string;
   "enable-embedding": boolean;
   "enable-nested-queries": boolean;
   "enable-query-caching"?: boolean;
-  "query-caching-ttl-ratio": number;
-  "query-caching-min-ttl": number;
   "enable-public-sharing": boolean;
   "enable-xrays": boolean;
-  "google-auth-auto-create-accounts-domain": string | null;
-  "google-auth-configured": boolean;
-  "jwt-enabled"?: boolean;
-  "jwt-configured"?: boolean;
-  "openai-api-key": string | null;
-  "openai-organization": string | null;
-  "openai-model": string | null;
-  "openai-available-models"?: OpenAiModel[];
-  "premium-embedding-token": string | null;
-  "saml-configured"?: boolean;
-  "saml-enabled"?: boolean;
   "search-typeahead-enabled": boolean;
-  "session-cookie-samesite": SessionCookieSameSite;
-  "show-database-syncing-modal": boolean;
   "show-homepage-data": boolean;
   "show-homepage-pin-message": boolean;
   "show-homepage-xrays": boolean;
   "site-uuid": string;
-  "slack-app-token": string | null;
-  "slack-files-channel": string | null;
-  "slack-token": string | null;
-  "slack-token-valid?": boolean;
   "subscription-allowed-domains": string | null;
-  "token-status": TokenStatus | null;
   "user-locale": string | null;
-  "version-info": VersionInfo | null;
   "uploads-enabled": boolean;
   "uploads-database-id": number | null;
   "uploads-schema-name": string | null;
   "uploads-table-prefix": string | null;
   "user-visibility": string | null;
+}
+
+interface AdminSettings {
+  "active-users-count"?: number;
+  "deprecation-notice-version"?: string;
+  "embedding-secret-key"?: string;
+  "query-caching-ttl-ratio": number;
+  "query-caching-min-ttl": number;
+  "google-auth-auto-create-accounts-domain": string | null;
+  "google-auth-configured": boolean;
+  "jwt-enabled"?: boolean;
+  "jwt-configured"?: boolean;
+  "premium-embedding-token": string | null;
+  "saml-configured"?: boolean;
+  "saml-enabled"?: boolean;
+  "show-database-syncing-modal": boolean;
+  "token-status": TokenStatus | null;
+  "version-info": VersionInfo | null;
```

Comment:
I don't think having a setting marked as potentially `undefined` is the best possible approach here. This interface is ambiguous in a sense that you don't know if it is going to be used in a setter or a getter.

If possible, we should strive to define types by looking only at the potential values for the particular setting. So when you use it in a setter, you don't risk allowing undefined.

And when used in a getter, the result of a fetch should be either Type or undefined.

Btw, this is not related to AdminSettings only.
It is a broader topic/problem that I think we need to solve.